File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstServiceProviders.java
package com.harry1453.burst.explorer.service;

import com.harry1453.burst.explorer.repository.ConfigRepository;

public final class BurstServiceProviders {
    public static BurstServiceProvider getBurstServiceProvider(ObjectService objectService, ConfigRepository configRepository) {
        return new BurstServiceProvider(objectService, configRepository);
    }

    public static ObjectService getObjectService(NetworkService networkService) {
        return new NetworkObjectService(networkService, new GsonDeserializerService());
    }

    public static ObjectService getObjectService(NetworkService networkService, DeserializerService deserializerService) {
        return new NetworkObjectService(networkService, deserializerService);
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstServiceProviders.kt
package com.harry1453.burst.explorer.service

import com.harry1453.burst.explorer.repository.ConfigRepository

object BurstServiceProviders {
    @JvmStatic
    fun getBurstServiceProvider(objectService: ObjectService, configRepository: ConfigRepository): BurstServiceProvider {
        return BurstServiceProvider(objectService, configRepository)
    }

    @JvmStatic
    fun getObjectService(networkService: NetworkService): ObjectService {
        return NetworkObjectService(networkService, GsonDeserializerService())
    }

    @JvmStatic
    fun getObjectService(networkService: NetworkService, deserializerService: DeserializerService): ObjectService {
        return NetworkObjectService(networkService, deserializerService)
    }
}
